### ISSUE TITLE
remove duplicate call to _get_weights_and_dims

### ISSUE THREAD

**Issue by matt-long** [NCAR/esmlab#87]
https://github.com/NCAR/esmlab/blob/878b80c3477a6c9fd990148ada299b51217d3020/esmlab/statistics.py#L147

Since this function calls `weighted_sum`, we're call `_get_weights_and_dims` twice.

We could change this
```python
    x_w_mean = weighted_sum(
        x, weights=weights, dim=op_over_dims, apply_nan_mask=apply_nan_mask_flag
    ) / weights.sum(op_over_dims)
```
to this
```python
x_w_mean = (x * weights).sum(op_over_dims) / weights.sum(op_over_dims)
```

**Comment by sudharsana-kjl** [comment]
#88

**Comment by andersy005** [comment]
Fixed in #88
